docs: update troubleshooting guide (#107)
autowarefoundation/autoware-documentation: docs/support/troubleshooting.md

# Troubleshooting

- [Setup script](#setup-script)
  - [CUDA-related errors](#cuda-related-errors)
- [Build errors](#build-errors)
  - [Insufficient memory](#insufficent-memory)
  - [Errors when using the latest version of Autoware](#errors-when-using-the-latest-version-of-autoware)
  - [Errors when using a fixed version of Autoware](#errors-when-using-a-fixed-version-of-autoware)
- [Docker/rocker issues](#dockerrocker-issues)
- [Runtime problems](#runtime-problems)

>   - [Map does not display when running the Planning Simulator](#map-does-not-display-when-running-the-planning-simulator)

Reviewer: I think it's not good to create such a table of contents for troubleshooting because the number of sections will gradually increase.

Also, we have the TOC on the right of the page.
![image](https://user-images.githubusercontent.com/31987104/168236036-ff9567fa-bece-43f9-a681-489009c3093f.png)

Author: Good point. I will remove it.

> ## Setup script

Reviewer: Better?

```suggestion
## Setup errors
```

Or
```suggestion
## Setup issues
```

Author: I went with `Setup issues` and made the change at the same time that I removed the TOC.

!!! warning

>     Note that some components in Autoware Universe require CUDA, and only CUDA 11.4 is supported at this time.

Reviewer: Mm, if possible, I don't want to write the version here because we'll update it soon.
Considering the maintenance cost, replacing this with a link is better.

Reviewer: Like this?

```suggestion
    Note that some components in Autoware Universe require CUDA, and only the CUDA version in the [env file](https://github.com/autowarefoundation/autoware/blob/main/amd64.env) is supported at this time.
```

Author: That fixes the problem perfectly!

I originally said something like `only the CUDA version stated in the installation guide is supported`, but thought that the version is only mentioned in the manual installation steps and that would also change in the future so decided to write it here instead, which just moved the problem to a different place. 😅

    Also, although Autoware could work with different versions of CUDA libraries as well,
    remember it is not officially supported and may sometimes be broken.

> ## Build errors

Reviewer: Maybe this also should be `issues`?

```suggestion
## Build issues
```

!!! warning

>     Note that some components in Autoware require CUDA.

Reviewer: Autoware Core shouldn't be dependent on CUDA. So I think it's good to specify Autoware Universe here.
What do you think?

Author: I apologise if my understanding is flawed, but I thought that Core was intended to be a stable version of Autoware that had been code reviewed, tested etc, and so if Universe required CUDA for perception then it was reasonable to assume that Core would require CUDA as well? 

I also seem to remember reading that there is a plan to remove the CUDA dependence in the future. Would that apply to Core only or Universe as well?

Reviewer: `Core` can be regarded as the refined version of `Universe`, but it includes dependency cleanup in addition to code quality improvements.
Since Autoware is desired to be hardware agnostic, we should avoid CUDA that is locked into NVIDIA.
Regarding `Universe`, I believe CUDA is allowed for more efficiency.

We should update the Autoware concepts documentation.
cc @mitsudome-r @xmfcx

Author: Okay, something like this?

```suggestion
    Note that some components in Autoware Universe require CUDA.
```

Reviewer: Yes, thank you!

    Also, although Autoware could work with different versions of CUDA libraries as well,
    remember it is not officially supported and may sometimes be broken.

## Build errors

> ### Insufficent memory

Reviewer: ```suggestion
### Insufficient memory
```

Author: A genuine spelling mistake. How embarrassing. 😅

> Building Autoware requires a lot of memory, and your machine can freeze or crash if memory runs out during a build. To avoid this problem, 16-32GB of swap should be configured. For details on how to configure a swap file, refer to [this Digital Ocean guide](https://www.digitalocean.com/community/tutorials/how-to-add-swap-space-on-ubuntu-20-04).

Reviewer: I think it's better to write the commands here instead of the link. @LalithVipulananthan What do you think?

```suggestion
Building Autoware requires a lot of memory, and your machine can freeze or crash if memory runs out during a build. To avoid this problem, 16-32GB of swap should be configured.

```bash
# Optional: Check the current swapfile
free -h

# Remove the current swapfile
sudo swapoff /swapfile
sudo rm /swapfile

# Create a new swapfile
sudo fallocate -l 32G /swapfile
sudo chmod 600 /swapfile
sudo mkswap /swapfile
sudo swapon /swapfile

# Optional: Check if the change is reflected
free -h
```
```

Reviewer: You can use some different commands, but I personally use the commands above.

Reviewer: Also, it's okay to keep the link as a reference.

### Insufficient memory

Building Autoware requires a lot of memory, and your machine can freeze or crash if memory runs out during a build. To avoid this problem, 16-32GB of swap should be configured.

```bash
# Optional: Check the current swapfile
free -h

# Remove the current swapfile
sudo swapoff /swapfile
sudo rm /swapfile

# Create a new swapfile
sudo fallocate -l 32G /swapfile
sudo chmod 600 /swapfile
sudo mkswap /swapfile
sudo swapon /swapfile

# Optional: Check if the change is reflected

> free -h

Reviewer: I'm sorry, maybe my suggestion was wrong? Missing ` ``` ` :thinking: 
``````suggestion
free -h
```
``````

Reviewer: I've manually fixed it.
